src/demo/components/common/IntentButton.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import {Payload} from '../../../client/context';
+
+interface IntentButtonProps {
+  title?: string;
+  iconClassName?: string;
+  action: () => Promise<void>;
+}
+
+enum ButtonStateType {
+  DEFAULT,
+  SPIN,
+  ERROR
+}
+
+export function IntentButton(props: IntentButtonProps): React.ReactElement {
+  const {action, title, iconClassName} = props;
+  const [buttonState, setButtonState] = React.useState(ButtonStateType.DEFAULT);
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
```

Comment:
Also need to `stopPropogation` on these events. Clicking one of these currently will both fire an intent and broadcast the context (which happens when clicking on a `SymbolsRow`/`ContactsRow`.

Reply:
Yep fixed that in my in another branch I'll be merging in tonight.